apps/api/e2e/utils/server.ts

```diff
@@ -18,167 +18,64 @@ const BASE_URL = `http://${HOST}:${PORT}${API_PREFIX ? `/${API_PREFIX}` : ""}`;
  *
  * @returns A promise that resolves to the HTTP server instance
  */
-export async function startServer(): Promise<Server> {
-  // First try to kill any existing servers on the test port
-  await killExistingServers();
+export async function startServer(): Promise<void> {
+  if (serverProcess) {
+    throw new Error("Cannot start test server twice");
```

Comment:
because we had no check here and the `startServer` function was being called more than once due to confusion on how setup files work, we were sometime trying to start servers repeatedly.  Because the port was in use this may have gone unnoticed.